docs/audit_report/src/side_channels/01_01_lms.rst

```diff
@@ -0,0 +1,101 @@
+"""""
+LMS
+"""""
+
+Analysed variants:
+
+- Truncated(SHA-256,192),HW(5,1)
+
+For the analysis of LMS, a utility was written that calls the functions to be analysed in a similar way to the Botan CLI.
+The following call is used to create the signature:
+
+.. code-block:: cpp
+
+    auto sk = Botan::HSS_LMS_PrivateKey(rng, mode);
+
+    Botan::PK_Signer sig(sk, rng, "");
+    signature = sig.sign_message(message, rng);
+
+
+**Leak: hash chain calculation**
+
+During the analysis, a control flow leak was found in the function `process()` [BOTAN_LMS_PROCESS]_.
+This function is part of the Winternitz One-Time Signature (WOTS) procedure.
+The WOTS procedure is based on the use of hash chains with a defined length.
+Using the private WOTS key, the initial values of the hash chain are calculated.
+These values are the private keys of the hash chains.
+The public keys of these hash chains are calculated by calculating the hash chain to the defined end.
+These public keys of the hash chains are used in a further calculation step to calculate the WOTS public key.
+To create a signature, the hash of a message and an additional checksum are split into individual parts, so-called chunks.
+Each chunk is signed by a hash chain.
+The number of steps in the hash chain calculation corresponds to the value of the respective chunk.
+The control flow leak indicates these differences in execution.
+
+.. code-block:: cpp
+
+    void process(HashFunction& hash,
+             uint16_t chain_idx,
+             uint8_t start,
+             uint8_t end,
+             std::span<const uint8_t> in,
+             std::span<uint8_t> out) {
+       BOTAN_ARG_CHECK(start <= end, "Start value is bigger than end value");
+
+       if(start == end) {
+          copy_into(out, in);
+          return;
+       }
+       m_gen.set_i(chain_idx);
+
+       // Unroll first iteration of the loop
+       m_gen.set_j(start++);
+       m_gen.gen(out, hash, in);
+
+       for(uint8_t j = start; j < end; ++j) {
+          m_gen.set_j(j);
+          m_gen.gen(out, hash, out);
+       }
+    }
+
+The variables *start* and *end* specify the start and the end index in the hash chain calculation.
+Depending on the value of the message hash, the *process()* routine is executed differently.
+This is the basis of the observed control flow leak.
```

Comment:
It would be nice if this could be explained a bit better.
In particular: does this only leak information about the message or what is leaked here. The remaining part of the section seems to say that the leak is unproblematic but, honestly, I don't get the details here.